PeerioTechnologies/peerio-icebear
# feat(files): download files with .peeriodownload suffix and then rename

Accidentally included the other PR, will rebase after that one is merged.

[PR guideline](https://github.com/PeerioTechnologies/peerio-icebear/blob/dev/docs/CONTRIBUTING.md)

### PR author

#### Relevant issue tracker links  

https://peerio.tpondemand.com/entity/7633

#### Linked PRs in other repos

Mobile rename for FileStream:
https://github.com/PeerioTechnologies/peerio-mobile/pull/68

#### Breaking changes  

None.

#### Manual testing instructions  

1. Download a file.
2. On desktop, watch the directory while it's being downloaded — see how it first appears as *filename*.peeriodownload.
3. Make sure the file is renamed to its original name (selected in Save dialog) after download finishes.

Do the same with a file that's already exists in the destination folder (when save dialog asks if you want to replace file). Make sure the file gets replaced.


#### Additional notes
... 

----
### Reviewer

* Localisation.
* Architectural decisions.
* Performance.
* Comments.
* Big feature tech design documentation.
* Readability.
* DRY.
* Are there any bicycles invented that should be replaced with existing code or external libs?
* Are there unit and e2e tests added?

----
### Repository owner

- [ ] Breaking changes are applied correctly.
